USE <db> + ALTER DATABASE COLLATE is not handled coorectly

### What did you do?

```
MySQL [(none)]> use all_mode;
Database changed
MySQL [all_mode]> alter database collate=utf8mb4_bin;
Query OK, 1 row affected (0.039 sec)
```

### What did you expect to see?

the DDL is parsed correctly and processed when the task has only allowed `all_mode` database to replicate

### What did you see instead?

```
[2024/08/21 16:00:52.170 +08:00] [WARN] [ddl.go:1260] ["skip event"] [task=test] [unit="binlog replication"] [event=query] [statement="ALTER DATABASE COLLATE = utf8mb4_bin"] ["query event context"="{schema: all_mode, originSQL: alter database collate=utf8mb4_bin, startLocation: position: (mysql-bin|000001.000003, 430), gtid-set: 3579e84e-5f91-11ef-8455-0242ac110003:1-6, endLocation: position: (mysql-bin|000001.000003, 535), gtid-set: 3579e84e-5f91-11ef-8455-0242ac110003:1-7, lastLocation: position: (mysql-bin|000001.000003, 535), gtid-set: 3579e84e-5f91-11ef-8455-0242ac110003:1-7, re-sync: , needHandleDDLs: , trackInfos: }"]
```

If we removed the filter and replicate all database & table, the error is

```
[2024/08/21 16:07:09.788 +08:00] [ERROR] [subtask.go:359] ["unit process error"] [subtask=test] [unit=Sync] ["error information"="ErrCode:10006 ErrClass:\"database\" ErrScope:\"downstream\" ErrLevel:\"high\" Message:\"startLocation: [position: (mysql-bin|000001.000003, 600), gtid-set: 3579e84e-5f91-11ef-8455-0242ac110003:1-7], endLocation: [position: (mysql-bin|000001.000003, 705), gtid-set: 3579e84e-5f91-11ef-8455-0242ac110003:1-8], origin SQL: [alter database collate=utf8mb4_bin]: execute statement failed: ALTER DATABASE COLLATE = utf8mb4_bin\" RawCause:\"Error 1046 (3D000): No database selected\" "]
```

### Versions of the cluster

master

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

--- a/dm/pkg/parser/common.go
+++ b/dm/pkg/parser/common.go
@@ -201,6 +201,10 @@ func SplitDDL(stmt ast.StmtNode, schema string) (sqls []string, err error) {
 	case *ast.AlterSequenceStmt:
 	case *ast.DropSequenceStmt:
 	case *ast.AlterDatabaseStmt:
+		if v.AlterDefaultDatabase {
+			v.AlterDefaultDatabase = false
+			v.Name = schemaName
+		}
 	case *ast.CreateDatabaseStmt:
 		v.IfNotExists = true
 	case *ast.DropDatabaseStmt:

--- a/dm/pkg/parser/common_test.go
+++ b/dm/pkg/parser/common_test.go
@@ -66,7 +66,8 @@ var testCases = []testCase{
 		[][]*filter.Table{{genTableName("s1", "")}},
 		[][]*filter.Table{{genTableName("xs1", "")}},
 		[]string{"DROP DATABASE IF EXISTS `xs1`"},
-	}, {
+	},
+	{
 		"alter database collate utf8mb4_general_ci",
 		[]string{"ALTER DATABASE `test` COLLATE = utf8mb4_general_ci"},
 		[][]*filter.Table{{genTableName("test", "")}},

--- a/dm/pkg/parser/common_test.go
+++ b/dm/pkg/parser/common_test.go
@@ -66,6 +66,12 @@ var testCases = []testCase{
 		[][]*filter.Table{{genTableName("s1", "")}},
 		[][]*filter.Table{{genTableName("xs1", "")}},
 		[]string{"DROP DATABASE IF EXISTS `xs1`"},
+	}, {
+		"alter database collate utf8mb4_general_ci",
+		[]string{"ALTER DATABASE `test` COLLATE = utf8mb4_general_ci"},
+		[][]*filter.Table{{genTableName("test", "")}},
+		[][]*filter.Table{{genTableName("xtest", "")}},
+		[]string{"ALTER DATABASE `xtest` COLLATE = utf8mb4_general_ci"},
 	},
 	{
 		"drop table `Ss1`.`tT1`",

